go-controller/pkg/crd/egressqos/v1/types.go

```diff
@@ -59,6 +59,7 @@ type EgressQoSRule struct {
 	// This field is optional, and in case it is not set the rule is applied
 	// to all egress traffic regardless of the destination.
 	// +optional
+	// +kubebuilder:validation:Format="cidr"
```

Comment:
wow didn't know there was actually a cidr validation in kubebuilder? I used to always refer to https://master.book.kubebuilder.io/reference/markers/crd-validation.html and in the past we used to have no way to add these for v4/v6 cidrs or ips.

Reply:
please add some test cases where you give wrong inputs and its indeed catching them and not allowing users to create wrong crds..

Reply:
I looked in that book as well and couldn't see the expanded "format" options so I was wondering how Ori / Andreas figured out that this option was available.

Reply:
@tssurya added some tests, not sure if in the right place though :sweat_smile: